### ISSUE TITLE
[Bug] Resolution of captured images is extremely low

### ISSUE THREAD

**Issue by pbirnzain** [Ethereal-Developers-Inc/OpenScan#37]
First of all, the app looks great so far! Thank you for all the work you've put into it.
It's fast, easy to use with a nice tutorial, and the edge detection works great.

I'd like to switch, but there's a problem with the image resolution:
It's always extremely low, as if only a thumbnail was used. This is both when adding pages using normal scan, and importing from gallery.

I'm using a Samsung Galaxy S20 (Android 11) and the OpenScan version currenly on Google Play.

**Comment by veejayts** [comment]
Unsure why this happens. We received two other similar reports.  By default, the image is compressed only by 10% to reduce storage occupied.

**Comment by pbirnzain** [comment]
Ok, let me know if there's anything I can do to help.

**Comment by JayFoxRox** [comment]
I'm getting the same issue. My image file in the openscan/files folder is 400x300 pixels, when the input image was 2560x1920 (taken by net.sourceforge.opencamera).

This is happening on OpenScan version from play store, on Samsung S3 Mini, running lineageos (not quite sure which version from top of my head).

**Comment by dogs-1985** [comment]
> Unsure why this happens. We received two other similar reports. By default, the image is compressed only by 10% to reduce storage occupied.

Hmm.. I also have same problem, after cropped the image, I feel like quality worst.

My device's resolution is 720x1280 (Set low for battery saver), I load photo from gallery (High quality), after cropped I can see the photo not very sharp.

Is it related to device's screen resolution? If the device's resolution high, app will keep more details, else quality will lost.

**Comment by veejayts** [comment]
Yeah, currently the way the image is being saved is weird. The resolution of the image being cropped is reduced to what is visible on the screen. If the original resolution of the image is preserved, the app crashes. We are working on a solution which should hopefully fix this bug.

**Comment by Alex-Esko** [comment]
@veejayts and @vikram0230
The app is great, especially the auto-cropping feature.
On a Samsung S10 there is the same issue with low resolution after cropping. This issue makes the whole app more or less useless.

**Comment by tiagoclc** [comment]
I found a solution!

At function setBitmap in file ScanFragment.java, just change this:

        this.original = scaledBitmap;
to
        this.original = original;

should look like this:

`   private void setBitmap(Bitmap original) {
        Bitmap scaledBitmap = scaledBitmap(original, sourceFrame.getWidth(), sourceFrame.getHeight());
        this.original = original;
        sourceImageView.setImageBitmap(scaledBitmap);
        Bitmap tempBitmap = ((BitmapDrawable) sourceImageView.getDrawable()).getBitmap();
        Map<Integer, PointF> pointFs = getEdgePoints(tempBitmap);
        polygonView.setPoints(pointFs);
        polygonView.setVisibility(View.VISIBLE);
        int padding = (int) getResources().getDimension(R.dimen.scanPadding);
        FrameLayout.LayoutParams layoutParams = new FrameLayout.LayoutParams(tempBitmap.getWidth() + 2 * padding, tempBitmap.getHeight() + 2 * padding);
        layoutParams.gravity = Gravity.CENTER;
        polygonView.setLayoutParams(layoutParams);
    }`

**Comment by tiagoclc** [comment]
If you want a better quality for B&W filter, just change from:

    Mat mbgra(info.height, info.width, CV_8UC4, pixels);
    // init our output image
    Mat dst = mbgra.clone();

    cvtColor(mbgra, dst, CV_BGR2GRAY);
//    float alpha = 2.2;
//    float beta = 0;
//    dst.convertTo(dst, -1, alpha, beta);

    threshold(dst,dst,0,255,THRESH_BINARY | THRESH_OTSU);


to:


    Mat mbgra(info.height, info, CV_8UC4, pixels);
    // init our output image
    Mat dst = mbgra.clone();

    cvtColor(mbgra, dst, CV_BGR2GRAY);
    float alpha = 1.8;
    float beta = -80;
    dst.convertTo(dst, -1, alpha, beta);

    //threshold(dst,dst,50,255,THRESH_BINARY | CV_THRESH_TRIANGLE);

    //threshold(mbgra,dst,100,255,THRESH_BINARY | THRESH_OTSU);

    threshold(mbgra,dst,100,255, THRESH_OTSU| CV_THRESH_BINARY);

**Comment by veejayts** [comment]
@tiagoclc Could you create a PR implementing this fix in [dev_v2.2.0](https://github.com/Ethereal-Developers-Inc/OpenScan/tree/dev_v2.2.0) branch?
